src/include/gurt/telemetry_producer.h

```diff
@@ -6,24 +6,18 @@
 #ifndef __TELEMETRY_PRODUCER_H__
 #define __TELEMETRY_PRODUCER_H__
 
+#include <gurt/telemetry_common.h>
+
 /* Developer facing server API to write data */
 void d_tm_set_counter(struct d_tm_node_t *metric, uint64_t value);
 void d_tm_inc_counter(struct d_tm_node_t *metric, uint64_t value);
-int d_tm_increment_counter(struct d_tm_node_t **metric, uint64_t value,
-			   const char *fmt, ...);
-int d_tm_record_timestamp(struct d_tm_node_t **metric, const char *fmt, ...);
-int d_tm_take_timer_snapshot(struct d_tm_node_t **metric, int clk_id,
-			     const char *fmt, ...);
-int d_tm_mark_duration_start(struct d_tm_node_t **metric, int clk_id,
-			     const char *fmt, ...);
-int d_tm_mark_duration_end(struct d_tm_node_t **metric, int err,
-			   const char *fmt, ...);
-int d_tm_set_gauge(struct d_tm_node_t **metric, uint64_t value,
-		   const char *fmt, ...);
-int d_tm_increment_gauge(struct d_tm_node_t **metric, uint64_t value,
-			 const char *fmt, ...);
-int d_tm_decrement_gauge(struct d_tm_node_t **metric, uint64_t value,
-			 const char *fmt, ...);
+void d_tm_record_timestamp(struct d_tm_node_t *metric);
+void d_tm_take_timer_snapshot(struct d_tm_node_t *metric, int clk_id);
+void d_tm_mark_duration_start(struct d_tm_node_t *metric, int clk_id);
+void d_tm_mark_duration_end(struct d_tm_node_t *metric);
+void d_tm_set_gauge(struct d_tm_node_t *metric, uint64_t value);
+void d_tm_increment_gauge(struct d_tm_node_t *metric, uint64_t value);
+void d_tm_decrement_gauge(struct d_tm_node_t *metric, uint64_t value);
```

Comment:
I would have used inc_gauge and dec_gauge to be consistent with counter.

Reply:
Good point. I'll change them.